server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -951,47 +949,39 @@ private static void updateReplicationStrategy(Settings.Builder settingsBuilder,
+        List<String> overriddenSettings = getRemoteStoreOverriddenSetting(requestSettings);
+        if (overriddenSettings.isEmpty() == false) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "Cannot override [%s] settings when [%s] is set to [%s].",
+                    String.join("][", overriddenSettings),
+                    CLUSTER_REMOTE_STORE_ENABLED_SETTING.getKey(),
+                    isRemoteStoreClusterEnabled
+                )
+            );
+        }
 
-            settingsBuilder.put(SETTING_REMOTE_STORE_ENABLED, true);
-            String remoteStoreRepo;
-            if (Objects.equals(requestSettings.get(INDEX_REMOTE_STORE_ENABLED_SETTING.getKey()), "true")) {
-                remoteStoreRepo = requestSettings.get(INDEX_REMOTE_STORE_REPOSITORY_SETTING.getKey());
-            } else {
-                remoteStoreRepo = CLUSTER_REMOTE_STORE_REPOSITORY_SETTING.get(clusterSettings);
-            }
-            settingsBuilder.put(SETTING_REMOTE_SEGMENT_STORE_REPOSITORY, remoteStoreRepo)
-                .put(
-                    SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY,
-                    requestSettings.get(
-                        INDEX_REMOTE_TRANSLOG_REPOSITORY_SETTING.getKey(),
-                        CLUSTER_REMOTE_TRANSLOG_REPOSITORY_SETTING.get(clusterSettings)
-                    )
-                );
+        if (isRemoteStoreClusterEnabled == true) {
+            settingsBuilder.put(SETTING_REMOTE_STORE_ENABLED, true)
+                .put(SETTING_REMOTE_SEGMENT_STORE_REPOSITORY, CLUSTER_REMOTE_STORE_REPOSITORY_SETTING.get(clusterSettings))
+                .put(SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY, CLUSTER_REMOTE_TRANSLOG_REPOSITORY_SETTING.get(clusterSettings));
         }
     }
 
-    private static boolean canCreateRemoteStoreIndex(Settings requestSettings) {
-        return (INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings) == false
-            || INDEX_REPLICATION_TYPE_SETTING.get(requestSettings).equals(ReplicationType.SEGMENT))
-            && (INDEX_REMOTE_STORE_ENABLED_SETTING.exists(requestSettings) == false
-                || INDEX_REMOTE_STORE_ENABLED_SETTING.get(requestSettings));
+    private static List<String> getRemoteStoreOverriddenSetting(Settings requestSettings) {
+        List<String> overriddenSettings = new ArrayList<>();
+        if (INDEX_REMOTE_STORE_ENABLED_SETTING.exists(requestSettings) == true) {
+            overriddenSettings.add(SETTING_REMOTE_STORE_ENABLED);
+        }
+        if (INDEX_REMOTE_SEGMENT_STORE_REPOSITORY_SETTING.exists(requestSettings) == true) {
+            overriddenSettings.add(SETTING_REMOTE_SEGMENT_STORE_REPOSITORY);
+        }
+        if (INDEX_REMOTE_TRANSLOG_REPOSITORY_SETTING.exists(requestSettings) == true) {
+            overriddenSettings.add(SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY);
+        }
+        return overriddenSettings;
```

Comment:
Style nitpick, but I think you can write this as the following which in my opinion is bit less verbose and more clear:

```
return Stream.of(INDEX_REMOTE_STORE_ENABLED_SETTING,
        INDEX_REMOTE_TRANSLOG_REPOSITORY_SETTING,
        INDEX_REMOTE_SEGMENT_STORE_REPOSITORY_SETTING)
    .filter(setting -> setting.exists(requestSettings))
    .map(Setting::getKey)
    .collect(toList());
```